### ISSUE TITLE
Invalid fix suggestion for comparing 2 strings until last '/'

### ISSUE THREAD

**Issue by SpyroTEQ** [kalessil/phpinspectionsea#1400]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v3.0.15  |
| Language level | Any (I'm on PHP 7.2 here) |

Current behaviour
--
I have two strings and I want to check if their begining matches, until their last '/' (actually, it's \ in my prod code, but that doesn't matter). So I'm using this test:
```php
substr($class1, 0, strrpos($class1, '\\')) === substr($class2, 0, strrpos($class2, '\\'));
```

PHP EA suggests me to use this instead:
```php
strpos($class1, substr($class2, 0, strrpos($class2, '\\'))) === 0
```

But this will not yield the same result, like with `$c1 = 'a/b/c/d';` and `$c2 = 'a/b/e';`. See this command-line test:

```php
C:\Users\Xenos>php -r "$c1 = 'a/b/c/d'; $c2 = 'a/b/e'; var_dump(substr($c1, 0, strrpos($c1, '/')), substr($c2, 0, strrpos($c2, '/')));"
Command line code:1:
string(5) "a/b/c"
Command line code:1:
string(3) "a/b"

C:\Users\Xenos>php -r "$c1 = 'a/b/c/d'; $c2 = 'a/b/e'; var_dump(strpos($c1, substr($c2, 0, strrpos($c2, '/'))));"
Command line code:1:
int(0)
```

In the 1st case, the var_dump shows that strings will not match, as expected: the part of $c1 and $c2 before their last `/` is `a/b/c` and `a/b` so they don't start the same way (in prod, I'm using this to check if 2 classes are in the same namespace; maybe there is a neater way, but it's only 1 use case of such kind of comparison).

In the 2nd case, with the suggested replacing code, the result is different, as it is actually checking whether `a/b/c/d` starts with `a/b` which is $c2 until its last `/` (here, it is). So tests are different.

Expected behaviour
--
PHP EA should not suggest this replacement at all, because it's not the same algorithm.

Environment details
---
N/A

**Comment by kalessil** [comment]
Fixed!
